eval/final/components.py

```diff
@@ -216,6 +216,9 @@ def branch_eval_summary_to_json(
 
     print(f"GPU Available: {gpu_available}, Using: {gpu_name}")
 
+    if batch_size.isdigit():
+        batch_size = int(batch_size)
+
```

Comment:
This will fail if the user sets the batch_size as an int instead of a str. This update will check for the correct type before continuing.   

```suggestion
    if type(batch_size) == str:
        if batch_size.isdigit():
             batch_size = int(batch_size)
        elif batch_size != "auto":
           raise ValueError("Only 'auto' is a valid string for 'batch_size'") 
```

Reply:
will do in a follow-up, we'll need this for other params as well